Карточная игра в ТГ

bimolvin opened the pull request:
Задачи были плотно взаимосвязаны, поэтому pr общий

### Задачи:

- **Волкова**: раздача и выбор карт: компонент карты, генерация колоды, рандомная раздача карт, отработка нажатий на карту: проверка активности
- **Попов**: логика игры: выбор карты пользователем, выбор карты ботом, проверка на победу в раунде, проверка на победу во всей игре
- **Борченко**: визуал: компоненты поля, руки игрока, карты; верстка, учёт мобильной версии
- **Бобовников**: отображение хода игры: статус хода, счётчик побед, кнопка начать заново, модальное окно с результатом игры, очистка поля по завершении раунда
- **Якимов**: развёртка в тг, получение данных пользователя*

*реализация самого приложения сильно затянулась, поэтому развёртки вовремя не случилось( до завтра появится


https://github.com/user-attachments/assets/367c89f8-ad6d-480a-9c53-f05320c1dbd6

bimolvin committed:
visual main

SomeExplorer committed:
Add deck generator

SomeExplorer committed:
Fix GameField

SomeExplorer committed:
Merge pull request #1 from wolloh/deck-builder

Add deck generator

bimolvin committed:
visual: mobile layout

bimolvin committed:
Merge pull request #2 from wolloh/mobile-layout

visual: mobile layout

wolloh committed:
[Engine] added base engine logic

wolloh committed:
[Engine] added engine store

wolloh committed:
Merge remote-tracking branch 'origin/master' into engine

SomeExplorer committed:
Connect engine with components

Jeinor committed:
Replaced score with gameScore in Player

Jeinor committed:
Added ResultGameModal for outputting results. Added score output and ResultGameModal to GameScores. Corrected the code in the engine.

wolloh committed:
[Engine] final changes

Jeinor committed:
Added a new game button. Minor changes.

reznikovkg commented on vue/src/components/App.vue:
```diff
@@ -1,91 +1,55 @@
 <template>
-  <div>
-    <RouterView />
```
Не ломайте логику роутера

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,101 @@
+<template>
+    <div div v-if="show">
```
Поправить отступы

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,101 @@
+            Результат игры
+        </h2>
+        <p class="winner">
+            {{this.getResultMessage}}
```
this убрать

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,101 @@
+            {{this.getResultMessage}}
+        </p>
+        <button class="restart-button" 
+        @click="restartGame">
```
Стрелочную

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,101 @@
+    ]),
+        getResultMessage() 
+        {
+            if (this.getEngine.winner === WinnerStates.PLAYER_WINS) {
```
Замените на объект

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,101 @@
+    },
+    
+    computed: {
+        ...mapGetters('game_engine', [
```
gameEngine

reznikovkg commented on vue/src/components/visual/ActiveBoard.vue:
```diff
@@ -0,0 +1,64 @@
+  }
+}
+
+@media(max-width: 560px) or (max-height: 580px){
```
Что это?

reznikovkg commented on vue/src/components/visual/GameCard.vue:
```diff
@@ -0,0 +1,181 @@
+  <div 
+    class="card" 
+    :class="{ 'card--face-down': faceDown, 'card--active': !faceDown && !onBoard }"
+    @click="makePlayerMove()"
```
Стрелочную

reznikovkg commented on vue/src/components/visual/GameCard.vue:
```diff
@@ -0,0 +1,181 @@
+    ])
+  },
+  methods: {
+    async makePlayerMove() {
```
Зачем?

reznikovkg commented on vue/src/components/visual/GameField.vue:
```diff
@@ -0,0 +1,52 @@
+      'getBotCards',
+      'getCardsOnBoard',
+    ]),
+    botCards() {
```
Можно не создавать эти поля, а сразу использовать поля из геттера (getBotCards)

reznikovkg commented on vue/src/components/visual/GameStatusHeader.vue:
```diff
@@ -0,0 +1,139 @@
+      'getEngine',
+    ]),
+    getCurrentTurn() {
+      let currentTurn = this.getEngine.turn == TurnStates.PLAYER 
```
В этом не было смысла и можно написать проще в шаблоне

reznikovkg commented on vue/src/engine/deck-builder.js:
```diff
@@ -0,0 +1,22 @@
+  }
+
+  generateDeck() {
+    let deck = []
```
const

reznikovkg commented on vue/src/core/utils.js:
```diff
@@ -0,0 +1,3 @@
+export async function sleep(ms) {
```
Лучше переделать на прямое использование по коду setTimeout + избавиться от лишних async/await

reznikovkg commented on vue/src/store/game_engine.js:
```diff
@@ -0,0 +1,14 @@
+        getPlayerCards: (state) => state.engine.player.cards,
+        getBotCards: (state) => state.engine.bot.cards,
+        getCardsOnBoard: (state) => state.engine.board,
+    }
```
Добавить мутации и экшены, вызывать методы движка из мутаций, мутации из экшены, а экшены из компонентов, не обращаться в методом движка, полученные через геттер

Jeinor committed:
ResultGameModal: arrow functions in @click event; Formatting. GameCar: arrow functions in @click event.

bimolvin committed:
add router, delete excess

bimolvin committed:
Merge branch 'master' of https://github.com/wolloh/webapp-tg

Jeinor committed:
ResultGameModal: added object instead of "if". GameStatusHeader: arrow functions in @click event.

SomeExplorer committed:
Improve engine store

This commit removes getEngine getter, adds actions and mutations and getters for some other fields. Also makes changes in components that use store.

SomeExplorer committed:
Merge pull request #3 from wolloh/engine-store

Improve engine store

SomeExplorer committed:
Fix getter in GameStatusHeader

reznikovkg commented:
Нужна демонстрация работы в боте

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,94 @@
+<template>
+  <div div v-if="show">
```
Что за див внутри?

Jeinor replied:
Случайно продублировал

reznikovkg commented on vue/src/components/modals/ResultGameModal.vue:
```diff
@@ -0,0 +1,94 @@
+<template>
+  <div div v-if="show">
+    <ModalComponent>
```
Принцип работы с модальными окнами неверный в принципе, В проекте есть пример запуска окон. И модуль для работы и запуска

Jeinor replied:
Я пытался сделать, как с HelpModal в оригинальном проекте, но он не работает

reznikovkg commented on vue/src/core/utils.js:
```diff
@@ -0,0 +1,3 @@
+export async function sleep(ms) {
+    await new Promise((resolve) => setTimeout(resolve, ms));
```
Почему не убрали?

bimolvin replied:
мы ещё в процессе, развернули пока в duck DNS: https://cardgameamm.duckdns.org/

reznikovkg commented on vue/src/engine/game-engine.js:
```diff
@@ -0,0 +1,119 @@
+
+        this.board.push(this.player.cards.find(card => card.id == cardId));
+
+        let playerCard = this.player.playCard(cardId);
```
let/const писать со смыслом

Jeinor committed:
Changing the way a modal window works

Jeinor committed:
fix name

wolloh committed:
change async/await to setTimeout and small refactor

wolloh committed:
Merge branch 'master' of https://github.com/wolloh/webapp-tg

wolloh committed:
fix bug

wolloh committed:
small refactor
